packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts

```diff
@@ -0,0 +1,85 @@
+import { openViewerTab, startTest } from "../helpers";
+import {
+  getSelectedTestCase,
+  getTestCaseSteps,
+  getTestRows,
+  getTestSuitePanel,
+  openPlaywrightTestPanel,
+} from "../helpers/testsuites";
+import { waitFor } from "../helpers/utils";
+import test, { expect } from "../testFixtureCloneRecording";
+
+test.use({ exampleKey: "playwright/breakpoints-05" });
+
+test("playwright-05: Test DOM node previews on user action step hover", async ({
+  pageWithMeta: { page, recordingId },
+  exampleKey,
+}) => {
+  await startTest(page, recordingId);
+  await openViewerTab(page);
+
+  await openPlaywrightTestPanel(page);
+
+  const testPanel = getTestSuitePanel(page);
+
+  const isVisible = await testPanel.isVisible();
+  expect(isVisible).toBe(true);
+
+  // has 1 test
+  const rows = getTestRows(page);
+  await waitFor(async () => {
+    await expect(rows).toHaveCount(1);
+  });
+
+  const breakpointsTest = rows
+    .filter({ hasText: "breakpoints-05: Test interaction of breakpoints with debugger statements" })
+    .first();
+
+  await breakpointsTest.click();
+  const selectedRow = getSelectedTestCase(page);
+  await waitFor(async () => {
+    expect(await selectedRow.isVisible()).toBe(true);
+    expect(selectedRow).toHaveCount(1);
+  });
+
+  const steps = getTestCaseSteps(selectedRow);
+  const firstStep = steps.first();
+
+  const userActionSteps = steps.filter({
+    hasText: /click/,
+  });
+
+  // Hovering over a user action step should show a preview of the DOM node
+  const lastClickStep = userActionSteps.last();
+  await lastClickStep.scrollIntoViewIfNeeded();
+
+  // Note that the ID is dynamically generated based on the piece of the box model shown
+  const highlighter = page.locator("#box-model-content");
+
+  await waitFor(
+    async () => {
+      // Repeatedly hover over the first step and then the click step, to force the
+      // `onMouseEnter` handler to keep checking if we have a DOM node entry available.
```

Comment:
Is there a better way we could wait for this case? I think we typically use data attributes to signal readiness for cases like this (and then we read them, and `delay(...)` before reading again if necessary)

Reply:
I don't think so. We don't use the DOM node cache at all while rendering. It's only checked in the `onHover` logic, synchronously, to see if we have it available.

Reply:
FWIW I think we could still do what I described here. The `useCacheStatus` hook (in "suspense") lets you peek at a cache status without kicking off a fetch.